internal/beater/beater.go

```diff
@@ -568,26 +575,14 @@ func maxConcurrentDecoders(memLimitGB float64) uint {
 	return decoders
 }
 
-// maxGroupsForAggregation calculates the maximum service groups that a
-// particular memory limit can have. This will be scaled linearly for bigger
-// instances.
-func maxGroupsForAggregation(memLimitGB float64) int {
-	const maxMemGB = 64
-	if memLimitGB > maxMemGB {
-		memLimitGB = maxMemGB
-	}
-	return int(memLimitGB * 1_000)
-}
-
-// maxTxGroupsForAggregation calculates the maximum transaction groups that a
-// particular memory limit can have. This will be scaled linearly for bigger
-// instances.
-func maxTxGroupsForAggregation(memLimitGB float64) int {
+// linearScaledValue calculates linearly scaled value based on memory limit using
+// the formula y = (perGBIncrement * memLimitGB) + c
+func linearScaledValue(perGBIncrement, memLimitGB, c float64) int {
```

Comment:
```suggestion
// linearScaledValue calculates linearly scaled value based on memory limit using
// the formula y = (perGBIncrement * memLimitGB) + constant
func linearScaledValue(perGBIncrement, memLimitGB, constant float64) int {
```

a more descriptive param name in the signature would be useful!